controllers/PostsService.js

```diff
@@ -38,6 +39,41 @@ const addUserData = async (user, post) => {
   }, post);
 };
 
+const getDefaultPosts = async ({ limit, page, query, sort }) => {
+  let postLimit = Math.ceil(limit / 2);
+  // get sources
+  const sources = await Source.list({
+    query: {
+      isDeleted: false,
+    },
+    select: '_id',
+  });
+
+  const promises = sources.map(async ({ _id }) => {
+    query.source = _id;
+
+    const post = await Post.list({
+      page,
+      sort,
+      query,
+      limit: postLimit,
+    });
+    postLimit = limit - postLimit;
+    return post;
+  });
+
+  const result = await Promise.all(promises);
+  const posts = [];
+
+  while (result[0].length > 0 || result[1].length > 0) {
+    if (result[0].length > 0) posts.push(result[0].pop());
+    if (result[1].length > 0) posts.push(result[1].pop());
+  }
+  posts.reverse().length = limit;
```

Comment:
because we already select enough number of posts of each source so no need to cut it.